### ISSUE TITLE
org-roam v2 should use sqlite3 queries to do filtering and sorting for org-roam-node-find and friends

### ISSUE THREAD

**Issue by cpbotha** [org-roam/org-roam#1835]
Currently, when you `org-roam-node-find`, the full list of nodes is read from the sqlite3 database via the following chain of function calls: `org-roam-node-find -> org-roam-node-read -> org-roam-node-read--completions -> org-roam-node-list` (org-roam-node-list dumps the whole node list, joined with other relevant tables)

Filtering and sorting then happens on the full list (in my case 2000 nodes, some folks have many more) in emacs-lisp (in my case nativecomp). Fast hardware and nativecomp, but keypress latency is noticeable.

Either by using zot's idea of sqlite3  fts (full text search) [1], or just by using the right sort of sqlite3 text INDEXes (will have to experiment with `LIKE` queries for this, but even a non-indexed query will be faster than emacs-lisp), it would be substantially faster to filter and and sort e.g. the first N hundred candidates directly from the database via sql query before showing in the completion list.

consult-grep's `#search-string-for-grep#refine-search-string-in-emacs` syntax is a great example of such a thing in practice, although generally with consult-grep the grep-search-string is already sufficient.

Long story short, with every character press a new query is executed by sqlite and returned to emacs. Much of this has already been discussed in zot's forum thread [1]

I am parking this feature request here in case I or someone else can make time to continue with zot's work, or implement something simpler.

Using sqlite3 in this way would enable working with thousands of nodes without much slowdown.

[1] Related discussion on the forum: https://org-roam.discourse.group/t/fast-and-scalable-sub-second-ivy-completion-for-v2-with-thousands-of-zettels/1584/18 (zot demonstrates sqlite3 fts searches)
